src/network/networkd-link.c

```diff
@@ -342,6 +343,12 @@ void link_update_operstate(Link *link) {
         else
                 operstate = LINK_OPERSTATE_OFF;
 
+        /* Even if the link does not have SLAVE flag, we will set it as 'enslaved' when it has no
+         * Address= setting and is joined to other netdevs. */
+        if (IN_SET(operstate, LINK_OPERSTATE_DEGRADED, LINK_OPERSTATE_CARRIER) &&
+            (link->flags & IFF_SLAVE || (link->enslaved && link->network && !link->network->static_addresses)))
```

Comment:
maybe turn this last last into a new mini helper macro/static inline function link_is_slave() or so?`I think that would make it more readable and hide the specific checks away a bit